crates/router/src/services/api.rs

```diff
@@ -1315,8 +1315,37 @@ pub fn build_redirection_form(
                         input type="hidden" name=(field) value=(value);
                     }
                 }
-
-                (PreEscaped(format!("<script type=\"text/javascript\"> {logging_template} var frm = document.getElementById(\"payment_form\"); window.setTimeout(function () {{ frm.submit(); }}, 300); </script>")))
+                (PreEscaped(format!(
+                    "<script type=\"text/javascript\"> {logging_template}
+                    var frm = document.getElementById(\"payment_form\"); 
+                    var formFields = frm.querySelectorAll(\"input\");
+                
+                    if (frm.method.toUpperCase() === \"GET\" && formFields.length === 0) {{
+                        window.setTimeout(function () {{
+                            window.location.href = frm.action;
+                        }}, 300);
+                    }} else {{
+                        window.setTimeout(function () {{
+                            frm.submit();
+                        }}, 300);
+                    }}
+                    </script>"
+                )))
+                (PreEscaped(r#"<script type="text/javascript"> {logging_template}
```

Comment:
I believe you've duplicated this, could you remove the older one which wasn't using the raw strings?

Reply:
yes sure

Reply:
@SanchithHegde when I am using use `#r` instead of `format!` `redirect_to_url` is not loading as expected in authorization flow
here's my code when I use `r#`
```suggestion
                (PreEscaped(r#"
                <script type="text/javascript"> {logging_template}
                    var frm = document.getElementById("payment_form"); 
                    var formFields = frm.querySelectorAll("input");
                
                    if (frm.method.toUpperCase() === "GET" && formFields.length === 0) {{
                        window.setTimeout(function () {{
                            window.location.href = frm.action;
                        }}, 300);
                    }} else {{
                        window.setTimeout(function () {{
                            frm.submit();
                        }}, 300);
                    }}
                    </script>
                    "#))

```

Reply:
You'd have to use `format!()` along with the raw string notation:

```rust
                (PreEscaped(format!(r#"
                <script type="text/javascript"> {logging_template}
                    var frm = document.getElementById("payment_form"); 
                    var formFields = frm.querySelectorAll("input");
                
                    if (frm.method.toUpperCase() === "GET" && formFields.length === 0) {{
                        window.setTimeout(function () {{
                            window.location.href = frm.action;
                        }}, 300);
                    }} else {{
                        window.setTimeout(function () {{
                            frm.submit();
                        }}, 300);
                    }}
                    </script>
                    "#)))
```